src/lib/storage/pos_list.hpp

```diff
@@ -64,10 +64,12 @@ struct PosList final : private pmr_vector<RowID> {
           [&]() {
             if (size() == 0) return true;
             const auto& common_chunk_id = (*this)[0].chunk_id;
-            return std::all_of(cbegin(), cend(),
-                               [&](const auto& row_id) { return row_id.chunk_id == common_chunk_id; });
+            if (common_chunk_id == INVALID_CHUNK_ID) return false;
+            return std::all_of(cbegin(), cend(), [&](const auto& row_id) {
+              return row_id.chunk_id == common_chunk_id && row_id.chunk_offset != INVALID_CHUNK_OFFSET;
+            });
           }(),
-          "Chunk was marked as referencing only a single chunk, but references more");
+          "PosList was marked as referencing a single chunk, but references more or the PosList contains NULL values");
```

Comment:
imo these two checks are rather independent and the null value test has nothing to do with the references_single_chunk property. but having a dedicated check for null values might be too cumbersome as well. 
also: from my understanding, posLists never contain values, so "NULL values" might be confusing. maybe better justin "invalid values"

Reply:
PosLists can also contain NULL values. Think of two tables that hold no NULL values being joined using a LEFT JOIN. This may produce NULL values on the right side. Those are not referenceable, as they have just been created. In these cases, the PosList contains NULL values.

Also pinging @Bensk1 because he looked into PosLists lately.

Reply:
That is related to #1983, isn't it? At some point we could add something like `references_only_NULLs`.

Reply:
Yes.